sdk/linux_kernel_drivers/xdma/libxdma.c

```diff
@@ -62,13 +64,38 @@ static unsigned int interrupt_mode;
 module_param(interrupt_mode, uint, 0644);
 MODULE_PARM_DESC(interrupt_mode, "0 - MSI-x , 1 - MSI, 2 - Legacy");
 
-static unsigned int enable_credit_mp;
+static unsigned int enable_credit_mp = 1;
 module_param(enable_credit_mp, uint, 0644);
-MODULE_PARM_DESC(enable_credit_mp, "Set 1 to enable creidt feature, default is 0 (no credit control)");
+MODULE_PARM_DESC(
+	enable_credit_mp,
+	"Set 0 to disable credit feature, default is 1 ( credit control enabled)");
 
 unsigned int desc_blen_max = XDMA_DESC_BLEN_MAX;
 module_param(desc_blen_max, uint, 0644);
-MODULE_PARM_DESC(desc_blen_max, "per descriptor max. buffer length, default is (1 << 28) - 1");
+MODULE_PARM_DESC(desc_blen_max,
+		 "per descriptor max. buffer length, default is (1 << 28) - 1");
+
+#define XDMA_PERF_NUM_DESC 128
+
+/* Kernel version adaptative code */
+#if KERNEL_VERSION(4, 19, 0) <= LINUX_VERSION_CODE
```

Comment:
karen, this will work with vanilla kernels.      challenge that Redhat/centos/amazonlinux may have backported some of these already.   did u get a chance to test on AmazonLinux1 and AmazonLinux2 ?

Reply:
I have not tested it on AmazonLinux, I will see if I can get access. 
On the Redhat/Centos kernels, I can certainly add a compatible header file and update the makefile to detect the distribution version. What are the distro versions that I should support? 6.x, 7.x ?

Reply:
Tested on AmazonLinux2.